src/plugins/vis_type_pie/public/pie_component.tsx

```diff
@@ -0,0 +1,342 @@
+      ),
+    [
+      legendPosition,
+      setColor,
+      bucketColumns,
+      visParams.palette.name,
+      visParams.distinctColors,
+      visData.rows,
+      props.uiState,
+    ]
+  );
+
+  const splitChartColumnAccessor = visParams.dimensions.splitColumn
+    ? getSplitDimensionAccessor(
+        services.fieldFormats,
+        visData.columns
+      )(visParams.dimensions.splitColumn[0])
+    : undefined;
+  const splitChartRowAccessor = visParams.dimensions.splitRow
+    ? getSplitDimensionAccessor(
+        services.fieldFormats,
+        visData.columns
+      )(visParams.dimensions.splitRow[0])
+    : undefined;
+
+  const splitChartDimension = visParams.dimensions.splitColumn
+    ? visData.columns[visParams.dimensions.splitColumn[0].accessor]
+    : visParams.dimensions.splitRow
+    ? visData.columns[visParams.dimensions.splitRow[0].accessor]
+    : undefined;
+
+  return (
+    <div className="pieChart__container" data-test-subj="visTypePieChart">
+      <div className="pieChart__wrapper">
+        <LegendToggle
+          onClick={toggleLegend}
+          showLegend={showLegend}
+          legendPosition={legendPosition}
+        />
+        <Chart size="100%">
+          <ChartSplit
+            splitColumnAccessor={splitChartColumnAccessor}
+            splitRowAccessor={splitChartRowAccessor}
+            splitDimension={splitChartDimension}
+          />
+          <Settings
+            debugState={window._echDebugStateFlag ?? false}
+            showLegend={showLegend}
+            legendPosition={legendPosition}
+            legendMaxDepth={visParams.nestedLegend ? undefined : 1}
+            legendColorPicker={legendColorPicker}
+            flatLegend={Boolean(splitChartDimension)}
+            tooltip={tooltip}
+            onElementClick={(args) => {
+              handleSliceClick(
+                args[0][0] as LayerValue[],
+                bucketColumns,
+                visData,
+                splitChartDimension,
+                splitChartFormatter
+              );
+            }}
+            legendAction={getLegendActions(
+              canFilter,
+              getLegendActionEventData(visData),
+              handleLegendAction,
+              visParams,
+              services.actions,
+              services.fieldFormats
+            )}
+            theme={chartTheme}
+            baseTheme={chartBaseTheme}
+            onRenderChange={onRenderChange}
+          />
+          <Partition
+            id="pie"
+            smallMultiples={SMALL_MULTIPLES_ID}
+            data={visData.rows}
+            valueAccessor={(d: Datum) => getSliceValue(d, metricColumn)}
+            percentFormatter={(d: number) => percentFormatter.convert(d / 100)}
+            valueGetter={
+              !visParams.labels.show ||
+              visParams.labels.valuesFormat === ValueFormats.VALUE ||
+              !visParams.labels.values
+                ? undefined
+                : 'percent'
+            }
```

Comment:
I'm not sure if the problem is here or somewhere else, but with the following set of instruction you can have be in a inconsistent state:
- turn on `Show labels`
- configure label position `Inside`
- update (labels are shown)
- turn off `Show labels`
- update
- labels keep showing

<img width="897" alt="Screenshot 2021-04-29 at 15 40 27" src="https://user-images.githubusercontent.com/1421091/116560071-53beb280-a901-11eb-9d05-580c93022184.png">

Reply:
Nice catch

Reply:
Fixed https://github.com/elastic/kibana/pull/83929/commits/12abdd25cbbeddc7e0621624e8f668d2f7c4496d